src/client/eppo-client.ts

```diff
@@ -475,6 +493,83 @@ export default class EppoClient implements IEppoClient {
     return { variation, action };
   }
 
+  private ensureNonContextualSubjectAttributes(
+    subjectAttributes: BanditSubjectAttributes,
+  ): Attributes {
+    let result: Attributes;
+    if ('numericAttributes' in subjectAttributes && 'categoricalAttributes' in subjectAttributes) {
```

Comment:
Would the `instanceOf` operator work here?

Since `BanditSubjectAttributes` has the same shape as a `Record`, a user could conceivably pass a dict that has `numericAttributes` and `categoricalAttributes` as keys. Very contrived and unlikely; yes.

Reply:
We can't as `ContextAttributes` is a _type_.

(We get TypeScript compilation error `TS2693: ContextAttributes only refers to a type, but is being used as a value here`).

I think it's ok--if they pass in `numericAttributes` and `categoricalAttributes` I think its fair to assume they are explicitly passing attributes with context.